Rearranged folder structure and updated contributing guide (#58)
Qiskit/qiskit_sphinx_theme: CONTRIBUTING.md

In addition to the general guidelines there are specific details for
contributing to qiskit_sphinx_theme, these are documented below.

> # Folder Structure

Reviewer: The heading levels seem a bit off to me - I'd have thought this was a level 2, under the top-level "Contributing" heading at the top of the page?

There are a few important subfolders to be aware of:

## `/qiskit_sphinx_theme`

> This subfolder contains the raw html and css files that are used to set the framework for Qiskit documentation web pages. The files in this folder will set the styles when a project calls the `qiskit_sphinx_theme` in it's `conf.py`, unless the project overrides the style with their own files.

Reviewer: ```suggestion
This subfolder contains the raw html and css files that are used to set the framework for Qiskit documentation web pages. The files in this folder will set the styles when a project calls the `qiskit_sphinx_theme` in its `conf.py`, unless the project overrides the style with their own files.
```

> E.g. if a qiskit project (with the `qiskit_sphinx_theme` installed) wants to override any of the themes files (such as the `layout.html`) that the `qiskit_sphinx_theme` provides it needs to have it's own version of that file (e.g. `layout.html`) stored in its `docs/_templates` folder.

Reviewer: ```suggestion
E.g. if a qiskit project (with the `qiskit_sphinx_theme` installed) wants to override any of the themes files (such as the `layout.html`) that the `qiskit_sphinx_theme` provides it needs to have its own version of that file (e.g. `layout.html`) stored in its `docs/_templates` folder.
```

## `/docs`

> This subfolder contains a scaled down version fo the Sphinx build that builds the documentation for the Qiskit repos. It pulls styles from the `/qiskit_sphinx_theme` subfolder. You can check any changes you are making to the `qiskit_sphinx_theme` by building the documentation (see running locally section) and opening the html files generated in `/docs/_build/html`

Reviewer: ```suggestion
This subfolder contains a scaled down version of the Sphinx build that builds the documentation for the Qiskit repos. It pulls styles from the `/qiskit_sphinx_theme` subfolder. You can check any changes you are making to the `qiskit_sphinx_theme` by building the documentation (see running locally section) and opening the html files generated in `/docs/_build/html`.
```

## `/docs/sphinx_guide`
This subfolder contains some example `.rst` files that show how to implement specific Sphinx features such as class documentaition, panels layouts, images etc.

# Run locally

> 1. Set up a python environment to work in
> 2. run `pip install .` (must do this every time)
> 3. run `sphinx-build -b html docs/ docs/_build/html` to build the documentation. This will create the `/docs/_build` subfolder where you can see the html files generated based on the `.rst` files in the `docs` folder and the styles pulled from the `qiskit_sphinx_theme`.

Reviewer: Is it not possible to install one in editable mode?

Also:

```suggestion
1. Set up a python environment to work in.
2. Run `pip install .` (must do this every time).
3. Run `sphinx-build -b html docs/ docs/_build/html` to build the documentation. This will create the `/docs/_build` subfolder where you can see the html files generated based on the `.rst` files in the `docs` folder and the styles pulled from the `qiskit_sphinx_theme`.
```

Author: i'm not sure about editable mode, i was just following Matthew's instructions 😄

In addition to the general guidelines there are specific details for
contributing to qiskit_sphinx_theme, these are documented below.

> ------

Reviewer: Are these vertical separators deliberate?  I'm not complaining if so, just wondering if they're meant to be there.

Author: oh yeah they're intentional! I love a line separator 😄

## Folder Structure

There are a few important subfolders to be aware of:

### `/qiskit_sphinx_theme`
This subfolder contains the raw html and css files that are used to set the framework for Qiskit documentation web pages. The files in this folder will set the styles when a project calls the `qiskit_sphinx_theme` in its `conf.py`, unless the project overrides the style with their own files.

E.g. if a qiskit project (with the `qiskit_sphinx_theme` installed) wants to override any of the themes files (such as the `layout.html`) that the `qiskit_sphinx_theme` provides it needs to have its own version of that file (e.g. `layout.html`) stored in its `docs/_templates` folder.

### `/docs`
This subfolder contains a scaled down version of the Sphinx build that builds the documentation for the Qiskit repos. It pulls styles from the `/qiskit_sphinx_theme` subfolder. You can check any changes you are making to the `qiskit_sphinx_theme` by building the documentation (see running locally section) and opening the html files generated in `/docs/_build/html`.

### `/docs/sphinx_guide`

> This subfolder contains some example `.rst` files that show how to implement specific Sphinx features such as class documentaition, panels layouts, images etc.

Reviewer: ```suggestion
This subfolder contains some example `.rst` files that show how to implement specific Sphinx features such as class documentation, panel layouts, images etc.
```

(I'm not sure if it's meant to be "panels, layouts" or "panel layouts")

In addition to the general guidelines there are specific details for
contributing to qiskit_sphinx_theme, these are documented below.

# Folder Structure

There are a few important subfolders to be aware of:

## `/qiskit_sphinx_theme`
This subfolder contains the raw html and css files that are used to set the framework for Qiskit documentation web pages. The files in this folder will set the styles when a project calls the `qiskit_sphinx_theme` in it's `conf.py`, unless the project overrides the style with their own files.

E.g. if a qiskit project (with the `qiskit_sphinx_theme` installed) wants to override the `layout.html` that the `qiskit_sphinx_theme` provides it needs to have it's own `layout.html` stored in its `docs/_templates` folder.

## `/docs`
This subfolder contains a scaled down version fo the Sphinx build that builds the documentation for the Qiskit repos. It pulls styles from the `/qiskit_sphinx_theme` subfolder. You can check any changes you are making to the `qiskit_sphinx_theme` by building the documentation (see running locally section) and opening the html files generated in `/docs/_build/html`

> ## `/docs/core_pages`

Reviewer 2: I'm a bit confused about this is this just for testing the theme? If we want to do a docs build with local changes applied it would probably be easier to install the theme from source and just clone `Qiskit/qiskit` and build the full docs in CI. It'll be a bit slower but it'll give a better picture than trying to figure out how to maintain a local copy of a subset of the out of tree pages

Author: I think for making changes to the theme it makes sense to keep a mini sphinx build here, just so we can quickly see any UI changes we're making for things like the top bar and side bar. and then i can also test out how potential changes in a `conf.py` would effect things without doing a v long build. For UI/styling changes it helps to have as quick a build as possible.

As for the core pages we talked about moving some of the main pages (contributors guide, deprecation policy etc.) out of the metapackage and into this repo so we can deploy them straight from here. I'm not sure yet if it would be better to rewrite them all in pure html, or have this mini sphinx build generate the html from the .rst files (maybe we can discuss the COs deployment stuff more when you're back from vacation?). In the meantime I just put the files here in their `.rst` form, but perhaps it would be better to move them out of the /docs subdir until we decide how to handle them 🤔
